### DISCUSSION TITLE
define trait for a arbitary struct

### DISCUSSION THREAD

**Discussion by majidbl** [actix/actix-web#1949] [Q&A] [answered]
how define trait for a arbitary struct as actix handler for use in app route and service

**Comment by robjtede** [comment, marked as answer]
The closest you can do is implementing `HttpServiceFactory` for use in `{App, Scope, Resource}::service`.

Admittedly, the docs are scarce for this trait so I recommend looking at the docs for `AppService` and implementations for Resource and Scope for some guidance.

- https://docs.rs/actix-web/3.3.2/actix_web/dev/trait.HttpServiceFactory.html
- https://docs.rs/actix-web/3.3.2/actix_web/dev/struct.AppService.html
